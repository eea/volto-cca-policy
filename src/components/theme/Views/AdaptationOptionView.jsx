import React from 'react';
import { HTMLField, ContentMetadata } from '@eeacms/volto-cca-policy/helpers';
import { Fragment } from 'react';

function AdaptationOptionView(props) {
  const { content } = props;

  let dataDisplay = [
    {
      type: 'TODO',
      field: 'TODO',
      section: 'ao_category',
      title: 'Category',
    },
    {
      type: 'TODO',
      field: 'TODO',
      section: 'ipcc_category',
      title: 'IPCC categories',
    },
    {
      type: 'HTMLField',
      field: 'stakeholder_participation',
      section: 'stakeholder_participation',
      title: 'Stakeholder participation',
    },
    {
      type: 'HTMLField',
      field: 'success_limitations',
      section: 'success_factors',
      title: 'Success and Limiting Factors',
    },
    {
      type: 'HTMLField',
      field: 'cost_benefit',
      section: 'costs_benefits',
      title: 'Costs and Benefits',
    },
    {
      type: 'HTMLField',
      field: 'legal_aspects',
      section: 'legal',
      title: 'Legal Aspects',
    },
    {
      type: 'HTMLField',
      field: 'implementation_time',
      section: 'implementation',
      title: 'Implementation Time',
    },
    {
      type: 'HTMLField',
      field: 'life_time',
      section: 'life_time',
      title: 'Life Time',
    },
  ];

  const findSection = (title) => {
    const found = dataDisplay.filter((item) => item.title === title);
    console.log("SEARCH", title);
    console.log("FOUND", found);
    if (found.length > 0) {
      console.log(found[0]);
      return found[0];
    }
    return null;
  };

  const sectionID = (title) => {
    const found = findSection(title);
    if (found === null) {
      return title;
    }
    return found.section;
  };

  return (
    <div className="adaptation-option-view">
      <div className="ui container">
        <div className="ui grid">
          <div className="row">
            <div className="nine wide column left-col">
              <div className="ui label">Adaptation option</div>
              <h1>{content.title}</h1>
              <HTMLField
                value={content.long_description}
                className="long_description"
              />

              {dataDisplay.length > 0 && (
                <>
                  <h5 className="Adaptation-option-selector">
                    Additional Details
                  </h5>
                  <ul>
                    {dataDisplay.map(
                      (data, index) =>
                        content.hasOwnProperty(data.field) && (
                          <li key={index}>
                            <a href="#ao_category">{data.title}</a>
                          </li>
                        ),
                    )}
                  </ul>
                </>
              )}

              <h5 className="Adaptation-option-selector">
                Additional Details Old
              </h5>
              <ul>
                <li>
                  <a href="#ao_category">Category</a>
                </li>
                <li>
                  <a href="#ipcc_category">IPCC categories</a>
                </li>
                <li>
                  <a href="#stakeholder_participation">
                    Stakeholder participation
                  </a>
                </li>
                <li>
                  <a href="#success_factors">Success and Limiting Factors</a>
                </li>
                <li>
                  <a href="#costs_benefits">Costs and Benefits</a>
                </li>
                <li>
                  <a href="#legal">Legal Aspects</a>
                </li>
                <li>
                  <a href="#implementation">Implementation Time</a>
                </li>
              </ul>

              <h5 className="Adaptation-option-selector">
                Reference information
              </h5>
              <ul>
                <li>
                  <a href="#websites">Websites</a>
                </li>
                <li>
                  <a href="#source">Source</a>
                </li>
              </ul>

              <h4>Adaptation Details</h4>
              <br />

              <div id={sectionID('Category')}>
                <p>
                  <i>Category</i>
                </p>
                {content.category
                  .map((item) => item.token)
                  .sort()
                  .map((cat, index) => (
                    <Fragment key={index}>
                      {cat}
                      <br />
                    </Fragment>
                  ))}
              </div>
              <br />

              <div id={sectionID('IPCC categories')}>
                <p>
                  <i>IPCC categories</i>
                </p>
                {content.ipcc_category
                  .map((item) => item.title)
                  .sort()
                  .join(', ')}
                {dataDisplay.length > 0 && (
                  <>
                    {dataDisplay.map(
                      (data, index) =>
                        content.hasOwnProperty(data.field) && (
                          <Fragment key={index}>
                            <br />

                            <div id={sectionID(data.title)}>
                              <p>
                                <i>{data.title}</i>
                              </p>
                              <HTMLField
                                value={content[data.field]}
                                className="long_description"
                              />
                            </div>
                          </Fragment>
                        ),
                    )}
                  </>
                )}
              </div>
              <hr />

              <h4>Reference information</h4>

              {content?.websites?.length > 0 && (
                <>
                  <h5>Websites</h5>

                  {content.websites.map((url) => (
                    <a key={url} href={url}>
                      {url}
                    </a>
                  ))}
                </>
              )}

              <h5>Source</h5>
              <HTMLField value={content.source} />
            </div>
            <div className="three wide column right-col">
              <div style={{}}>
                <ContentMetadata {...props} />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default AdaptationOptionView;
