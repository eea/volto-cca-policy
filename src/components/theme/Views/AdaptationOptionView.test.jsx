import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-intl-redux';
import AdaptationOptionView from './AdaptationOptionView';
import renderer from 'react-test-renderer';

const mockStore = configureStore();

jest.mock('semantic-ui-react', () => ({
  ...jest.requireActual('semantic-ui-react'),
}));

describe('AdaptationOptionView', () => {
  it('should render the component', () => {
    const content = {
      title: 'AdaptationOptionView',
      geochars:
        '{\r\n "geoElements":{"element":"GLOBAL",\r\n "macrotrans":null,"biotrans":null,"countries":[],\r\n "subnational":[],"city":""}}',
      ipcc_category: [
        {
          title: 'Structural and physical: Ecosystem-based adaptation options',
          token: 'STRUCT_ECO',
        },
      ],
      websites: ['https://my-website.com'],
      source: {
        'content-type': 'text/html',
        data: '<p>Some text</p>',
        encoding: 'utf-8',
      },
    };

    const store = mockStore({
      userSession: { token: '1234' },
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <AdaptationOptionView content={content} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
