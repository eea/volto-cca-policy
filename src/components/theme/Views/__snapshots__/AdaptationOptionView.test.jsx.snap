// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptationOptionView should render the component 1`] = `
<div
  className="db-item-view adaptation-option-view"
>
  <div>
    Title Block Component
  </div>
  <div
    className="ui container"
  >
    <div
      className="ui twelve column grid"
    >
      <div
        className="row"
      >
        <div
          className="eight wide computer twelve wide mobile twelve wide tablet column col-left"
        >
          <h2>
            Description
          </h2>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
          <div
            className="adaptation-details"
          >
            <div>
              <h3>
                Additional Details
              </h3>
              <ul>
                <li>
                  <a
                    href="#ipcc_category"
                  >
                    IPCC categories
                  </a>
                </li>
              </ul>
            </div>
            <div>
              <h3>
                Reference information
              </h3>
              <ul>
                <li>
                  <a
                    href="#websites"
                  >
                    Websites
                  </a>
                </li>
                <li>
                  <a
                    href="#source"
                  >
                    Source
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="ui divider"
          />
          <h2>
            Adaptation Details
          </h2>
          <div
            className="section"
            id="ipcc_category"
          >
            <h5
              className="section-title"
            >
              IPCC categories
            </h5>
            Structural and physical: Ecosystem-based adaptation options
          </div>
          <div
            className="ui divider"
          />
          <h2>
            Reference information
          </h2>
          <h5
            id="websites"
          >
            Websites:
          </h5>
          <ul>
            <li>
              <a
                href="https://my-website.com"
              >
                https://my-website.com
              </a>
            </li>
          </ul>
          <div
            className="section"
            id="source"
          >
            <h5
              id="source"
            >
              References:
            </h5>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Some text</p>",
                }
              }
            />
          </div>
          <div
            className="published-modified-info"
          >
            <p>
              <span>
                <strong>
                  Published in Climate-ADAPT
                </strong>
                 
                Invalid Date
              </span>
              <span>
                   -   
              </span>
              <span>
                <strong>
                  Last Modified in Climate-ADAPT
                </strong>
                 
                Invalid Date
              </span>
            </p>
          </div>
          <div
            className="share-info"
          >
            <a
              href="/help/share-your-info"
            >
              <button
                className="ui button primary icon labeled"
              >
                Share your information
                <i
                  aria-hidden="true"
                  className="arrow right icon"
                />
              </button>
            </a>
          </div>
        </div>
        <div
          className="four wide computer twelve wide mobile twelve wide tablet column col-right"
        >
          <div
            className="ui segment"
          >
            <div
              className="content-metadata"
            >
              <h5>
                Geographic characterisation:
              </h5>
              <div
                className="geochar"
              >
                <p>
                  Global
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
